### ISSUE TITLE
Allow setting of email alert thresholds via API

### ISSUE THREAD

**Issue by naegelin** [linode/developers#182]
Current email alert thresholds can only be set manually making it very difficult for customers that have hundreds of linodes to update those thresholds. This issue is made worse by the fact that **email thresholds for CPU settings are set to a percentage of a single core**, meaning on a 4 core system, 25% cpu usage (100% of One core) will set off an email alert for  a threshold set to 90% (default threshold). This results in hundreds of useless emails daily for us and I'm sure other customers as well.

1) Expose email threshold settings per linode via api
2) Accurately calculate CPU threshold based on core count
3) Send hundreds of fewer email alerts daily
4) Increased profit for linode ;)

**Comment by eatonphil** [comment]
Hey! Actually, you are able to set alert thresholds in both the APIv3 (the production API) and the APIv4 (the alpha API). [Here are the docs](https://www.linode.com/api/linode/linode.update) for the APIv3 `linode.update` action that allows you to modify the alert settings. [Here are the docs](https://developers.linode.com/reference/#object-linode) for the APIv4 (alpha).

As far as the calculation of percentages go, my understanding is that we report them the same way standard Linux tools (like top) report them. [Here is a StackExchange answer](http://superuser.com/a/457634/173239) that explains how / why top works in that regard. Currently there is not a way to alert based on per-core settings.

**Comment by naegelin** [comment]
You are correct in that you report like top does but as your article link explains 90% means 90% of a single core. So on an 8 core machine 720% would actually be '90% of system utilization'. Its not very clear in your UI or API docs when you describe it as: 

`CPU Usage - % - Average CPU usage over 2 hours exceeding this value triggers this alert. (Range 0-2000, default 90)`

You will also notice that the range is 2000 which of course implies 100% of 20 cores but that might be adding additional confusion. 

Regardless I'm thrilled to see we can fix this and stop getting hundreds of email alerts daily when 1 of 8 cores is busy ;)

**Comment by eatonphil** [comment]
Thanks again for the feedback and we'll keep your thoughts in consideration. We're going to be taking this repo private in a little while. But if you'd like to make further issues, please see our primary repo: https://github.com/linode/manager. Thank you!
